### DISCUSSION TITLE
CUTLASS Integration

### DISCUSSION THREAD

**Discussion by mnicely** [Oneflow-Inc/oneflow#9326] [Ideas]
First off, it's very exciting to see CUTLASS being used in Oneflow! It's not clear if the fMHA example 42 is being used directly or if CUTLASS is a submodule. I wanted to let you know improvements to fMHA will released in 2.11 later this quarter.

**Comment by jackalcooper** [comment]
@mnicely Hi Matthew, very glad to hear from you! CUTLASS is such an amazing project!
 fMHA example 42's header is used in this file:
https://github.com/Oneflow-Inc/oneflow/blob/b2091bc9c58b1052708782d883956b433dbf26a6/oneflow/user/kernels/fused_multi_head_attention_inference_kernel.cu#L22
